MultiYearStudies.R

```diff
@@ -1,44 +1,53 @@
 
-library(skimr)
 
-#duration
-meta_df %>%
-  group_by(Study_number, OtherFactor, Ecosystem_type) %>%
-  mutate(Duration = Study_midyear - min(Study_midyear)) -> meta_time
+#conduct meta-analysis
+#this is the best fitting model of irrigiation studies - not sure if the "/" notation is correct
+#does it makes sense to have Ecosystem_type as a random effect???? Maybe???
 
 
-#Not working at the moment
-meta_time %>%
-  select(-Study_number, -Record_number, -OtherFactor, -Manipulation_level, -Soil_type) %>%
-  skim() -> summary
 
+res_irrigation <- rma(yi, vi, random = ~ 1|Study_number/Ecosystem_type,
+              mods = ~Duration*Percent_control*Latitude,
+              method = "ML",
+           data = metadat[metadat$Manipulation == "Irrigation",],
+           test = "knha")
+summary.rma(res_irrigation)
+profile(res_irrigation)
+
+
+res_drought <- rma.mv(yi, vi, random = ~ 1|Study_number/Ecosystem_type,
+                         mods = ~Duration * Percent_control * Latitude,
+                         method = "ML",
+                         data = metadat_short[metadat_short$Manipulation == "Drought",])
+summary.rma(res_drought)
+profile(res_drought)
+
+library(MuMIn)
+if (!require("devtools")) {
+  install.packages("devtools")
+}
+devtools::install_github("MathiasHarrer/dmetar")
+
+multimodel.inference(TE = "yi", seTE = "vi",
+                     data = metadat[metadat$Manipulation == "Drought",],
+                     predictors = c("Ecosystem_type", "Duration", "Percent_control", "Soil_drainage"),
+                     interaction = TRUE)
+
+#would be nice to drop desert and wetland, as those are under represented
+#this didn't actually reduce the AIC very much
+metadat %>%
+  filter(Ecosystem_type != c("Wetland", "Desert")) -> metadat_short
 
 
-#compute effect sizes
-metadat <- escalc(measure = "ROM",
-                  m1i = Manip_Resp, m2i = Control_Resp, 
-                  sd1i = Manip_SD, sd2i = Control_SD,
-                  n1i = N, n2i = N, 
-                  slab = paste(Study_number, Author, Study_midyear),
-                  data = meta_time)
 
-#conduct meta-analysis
-#check the call arguments for rma.mv vs rma
-res <- rma.mv(yi, vi, random = ~1|Study_number,
-           mods= ~Ecosystem_type + Duration + Percent_control - 1,
-           data = metadat[metadat$Manipulation == "Irrigation",])
-res
 
 #conduct meta-analysis
 res2 <- rma(yi, vi, 
-           mods= ~Ecosystem_type + Duration + Percent_control - 1,
+           mods= ~Ecosystem_type*Duration*Percent_control - 1,
```

Comment:
Put spaces around asterisks please, for readability.

Reply:
Done.